std/string.d

```diff
@@ -2468,7 +2468,7 @@ if ((hasSlicing!Range && hasLength!Range) ||
     static struct Result
     {
     private:
-        Range _input;
+        StringTypeOf!Range _input;
```

Comment:
This breaks lineSplitter for non-string ranges, e.g. byUTF!dchar.
What you want is an overload for the StringTypeOf handling.